File: api/src/main/java/no/difi/meldingsutveksling/api/StatusStrategy.java
package no.difi.meldingsutveksling.api;

import no.difi.meldingsutveksling.ServiceIdentifier;
import no.difi.meldingsutveksling.status.Conversation;
import no.difi.meldingsutveksling.status.MessageStatus;

import java.util.Set;

public interface StatusStrategy {

    void checkStatus(Set<Conversation> conversations);
    ServiceIdentifier getServiceIdentifier();
    boolean isStartPolling(MessageStatus status);
    boolean isStopPolling(MessageStatus status);

}

File: api/src/main/kotlin/no/difi/meldingsutveksling/api/StatusStrategy.kt
package no.difi.meldingsutveksling.api

import no.difi.meldingsutveksling.ServiceIdentifier
import no.difi.meldingsutveksling.status.Conversation
import no.difi.meldingsutveksling.status.MessageStatus

interface StatusStrategy {
    fun checkStatus(conversations: MutableSet<Conversation>)
    fun getServiceIdentifier(): ServiceIdentifier
    fun isStartPolling(status: MessageStatus): Boolean
    fun isStopPolling(status: MessageStatus): Boolean
}